### PR TITLE
Rebuild libxslt 1.1.37 with libxml2 2.10

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Update because of libxml2 2.10.3

Changelog: https://gitlab.gnome.org/GNOME/libxslt/-/blob/master/NEWS
License: https://gitlab.gnome.org/GNOME/libxslt/-/blob/master/Copyright
Requirements: 
- https://gitlab.gnome.org/GNOME/libxslt/-/blob/v1.1.37/INSTALL
- https://gitlab.gnome.org/GNOME/libxslt/-/blob/v1.1.37/configure.ac
- https://gitlab.gnome.org/GNOME/libxslt/-/blob/master/CMakeLists.txt

Actions:
1.  Add 0001-Make-and-install-a-pkg-config-file-on-Windows.patch for win
2. Remove ignore_run_exports:  icu  # [osx]
3. Add pkg-config to build
4. Pin libxml2 to 2.10 in host by the local cbc.yaml
5. Replace conda-build in test/requires (it causes errors) with pkg-config

### CODE REVIEW

**Review by M-Waszkiewicz-Anaconda — COMMENTED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,41 +7,40 @@ package:
...
-    - conda inspect objects -p $PREFIX libxslt  # [osx]
+    - pkg-config --cflags libxslt libexslt
+    - pkg-config --libs libxslt libexslt
 
 about:
   home: https://gitlab.gnome.org/GNOME/libxslt
   license: MIT
-  licence_family: MIT
+  license_family: MIT
   license_file: COPYING
```
Comment (akabanovs) [inline recipe/meta.yaml:44]: The license file name is Copyright, I think?

Comment (skupr-anaconda) [reply recipe/meta.yaml:44]: You're right. Fixed it

Comment (skupr-anaconda) [reply recipe/meta.yaml]: It was from the conda-forge recipe

**Review by akabanovs — APPROVED** [review]
LGTM

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+libxml2:
+  - 2.10
```
Comment (akabanovs) [inline recipe/conda_build_config.yaml:2]: Do we need recipe's cbc? Can we explicitly specify the version in the host section?

**Review by akabanovs — APPROVED** [review]
LGTM

**Review by danpetry — APPROVED** [review]
LGTM. As an aside, it would be nice to list in the description the reasons for the changes, as well as the changes themselves. You can see from the diff what has been changed, so it's more or less duplicated information to just list the changes.

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**